### PR TITLE
Add SoftDeletable concern

### PR DESCRIPTION
**Pull request by ozvillafan**

### CODE REVIEW

File: app/models/concerns/nunchaku/soft_deletable.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    end
+
+    protected
+
+    def rename
+      assign_attributes(name_field => "#{send(name_field)} - DELETED") if name_unique?
+    end
+
+    def soft_delete
+      self.is_deleted = true
```
Comment (mbandrewfoster) [inline app/models/concerns/nunchaku/soft_deletable.rb]: update_attributes!(:is_deleted => true) would be better here?

File: app/models/concerns/nunchaku/soft_deletable.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    def delete
+      soft_delete unless new_record?
+    end
+
+    def name_unique?
+      false
+    end
+
+    def name_field
```
Comment (mbandrewfoster) [inline app/models/concerns/nunchaku/soft_deletable.rb]: name_column would be more standard

File: app/models/concerns/nunchaku/soft_deletable.rb (hunk 1 above)
Comment (ozvillafan) [inline app/models/concerns/nunchaku/soft_deletable.rb]: It's calling other destroy callbacks. If they change the model (such as the rename) then it effectively does 2 updates with update_attributes!

File: app/models/concerns/nunchaku/soft_deletable.rb (hunk 2 above)
Comment (ozvillafan) [inline app/models/concerns/nunchaku/soft_deletable.rb]: will change shortly

**Pull request merged**